modules/dag-l1/src/main/scala/org/tessellation/dag/l1/storage/TransactionStorage.scala

```diff
@@ -0,0 +1,111 @@
+package org.tessellation.dag.l1.storage
+
+import cats.data.{NonEmptyList, NonEmptySet}
+import cats.effect.{Ref, Sync}
+import cats.syntax.applicative._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.option._
+
+import scala.collection.immutable.SortedSet
+
+import org.tessellation.dag.transaction.filter.Consecutive
+import org.tessellation.ext.collection.MapRefUtils._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{Transaction, TransactionReference}
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+import io.chrisdavenport.mapref.MapRef
+
+trait TransactionStorage[F[_]] {
+  def isParentAccepted(tx: Transaction): F[Boolean]
```

Comment:
is it required to pass a transaction here? maybe just a hash would be enough

Reply:
I think it's better to pass transaction, extract parent here and return info if the parent of this transaction is accepted. I think it's better to extract here in one place then to extract outside of it. If we would pass hash only, we would need to change the method to `isTransactionAccepted` and it's a different functionality.